src/java.base/share/classes/com/sun/crypto/provider/SunJCE.java

```diff
@@ -292,7 +295,12 @@ void putEntries() {
         attrs.put("SupportedModes", "GCM");
         attrs.put("SupportedKeyFormats", "RAW");
 
-        if (useNativeGaloisCounterMode) {
+        if (useNativeGaloisCounterMode
+/*[IF CRIU_SUPPORT]*/
+            && !InternalCRIUSupport.isCheckpointAllowed()
```

Comment:
This is problematic: I think it disables the native implementation even after a restore because the Java implementation will be registered before the checkpoint (both GCM and ChaCha20).

Reply:
There is no problem, `CRIUSECProvider` replaces other providers before taking checkpoint, and put them back after restore, all provider instances have to be recreated, the previously registered service will have been removed and re-registered. See `CRIUConfigurator.setCRIUSecMode()/setCRIURestoreMode()`.

Reply:
If this provider is not used before a checkpoint, then why does it need to ask `isCheckpointAllowed()`?